src/api/app/controllers/trigger_workflow_controller.rb

```diff
@@ -0,0 +1,47 @@
+class TriggerWorkflowController < TriggerController
+  skip_before_action :set_project, :set_package, :set_object_to_authorize, :set_multibuild_flavor
+  before_action :set_scm_event
+  before_action :validate_scm_event
+
+  def create
+    authorize @token
+    @token.user.run_as do
+      @token.call(scm: scm, event: event, payload: payload)
+      render_ok
+    end
+  end
+
+  private
+
+  def set_scm_event
+    @gitlab_event = request.env['HTTP_X_GITLAB_EVENT']
+    @github_event = request.env['HTTP_X_GITHUB_EVENT']
+  end
+
+  def validate_scm_event
+    raise BadScmHeaders unless @gitlab_event.present? || @github_event.present?
+  end
+
+  def scm
+    if @gitlab_event
+      'gitlab'
+    elsif @github_event
+      'github'
+    end
```

Comment:
What if this is neither?

Reply:
Before we call this method, there is a `before_action :validate_scm_event` that checks that, at least, one of the headers is present. So it will never be neither.